File: RxJavaExample/app/src/main/java/com/example/velm/rxjavaexample/MainActivity.java
package com.example.velm.rxjavaexample;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

import rx.Observable;
import rx.Observer;
import rx.Subscription;

public class MainActivity extends AppCompatActivity {

    private Subscription subscription;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Observable<Integer> observable = Observable.just(1,2,3,4,5);

        Observer<Integer> observer = new Observer<Integer>() {
            @Override
            public void onCompleted() {
                print("Completed");
            }

            @Override
            public void onError(Throwable e) {
                print(e.getMessage());
            }

            @Override
            public void onNext(Integer integer) {
               print(integer+"");
            }
        };

        subscription = observable.subscribe(observer);

        print(subscription.toString());
    }

    public void print(String message){
        Log.d("MainActivity",message);
        //Toast.makeText(getApplicationContext(),"value = "+message,Toast.LENGTH_SHORT).show();
    }


    @Override
    protected void onDestroy() {
        super.onDestroy();
        subscription.unsubscribe();
    }
}

File: MotionLayoutAndroid/app/src/main/java/com/velmurugan/motionlayoutandroid/MainActivity.kt
package com.velmurugan.motionlayoutandroid

import android.os.Bundle
import android.os.Handler
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.AppCompatButton
import androidx.constraintlayout.motion.widget.MotionLayout

class MainActivity : AppCompatActivity() {



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<AppCompatButton>(R.id.button1).show()

        Handler().postDelayed(
            Runnable {
                findViewById<MotionLayout>(R.id.motionLayout).transitionToEnd()
            }
        , 3000)

        val motionLayout = findViewById<MotionLayout>(R.id.motionLayout)
        motionLayout.addTransitionListener(object : MotionLayout.TransitionListener{
            override fun onTransitionStarted(p0: MotionLayout?, p1: Int, p2: Int) {

            }

            override fun onTransitionChange(p0: MotionLayout?, p1: Int, p2: Int, p3: Float) {

            }

            override fun onTransitionCompleted(p0: MotionLayout?, p1: Int) {

                when(p1) {

                    R.id.end -> {
                        motionLayout.setTransition(R.id.button_trans)
                        motionLayout.transitionToEnd()
                    }

                    R.id.button_end -> {
                        motionLayout.setTransition(R.id.trsn3)
                        motionLayout.postDelayed( Runnable {
                            motionLayout.transitionToEnd()
                        }
                            , 2000
                        )
                    }
                }

            }

            override fun onTransitionTrigger(p0: MotionLayout?, p1: Int, p2: Boolean, p3: Float) {

            }
        })


    }



    fun AppCompatButton.show() {
        Toast.makeText(this@MainActivity, "Message", Toast.LENGTH_SHORT).show()
    }

}